packages/oauth/oauth-provider/src/token/token-manager.ts

```diff
@@ -55,29 +51,45 @@ export class TokenManager {
     return new Date(now.getTime() + this.tokenMaxAge)
   }
 
-  protected async buildAccessToken(
+  protected async createAccessToken(
     tokenId: TokenId,
-    account: Account,
     client: Client,
+    account: Account,
     parameters: OAuthAuthorizationRequestParameters,
-    createdAt: Date,
+    issuedAt: Date,
     expiresAt: Date,
-    scope: string,
+    scope: OAuthScope,
   ): Promise<OAuthAccessToken> {
-    return this.signer.createAccessToken({
+    const claims: TokenClaims = {
       jti: tokenId,
       sub: account.sub,
+      iat: dateToEpoch(issuedAt),
       exp: dateToEpoch(expiresAt),
-      iat: dateToEpoch(createdAt),
-      cnf: parameters.dpop_jkt ? { jkt: parameters.dpop_jkt } : undefined,
+      aud: account.aud,
+      scope,
+      // https://datatracker.ietf.org/doc/html/rfc8693#section-4.3
+      client_id: client.id,
+
+      ...(parameters.dpop_jkt && {
+        cnf: { jkt: parameters.dpop_jkt },
+      }),
 
       ...(this.accessTokenMode === AccessTokenMode.stateless && {
         aud: account.aud,
-        scope,
+        scope: scope ?? parameters.scope,
```

Comment:
I believe `scope` is always a string (so never nullish) and also included above.  Can we nix it?
```suggestion
```